packages/admin/cms-admin/src/generator/future/generator.ts

```diff
@@ -17,7 +18,7 @@ export type FormFieldConfig<T> = (
     | { type: "staticSelect"; values?: string[] }
     | { type: "asyncSelect"; values?: string[] }
     | { type: "block"; block: BlockReference }
-) & { name: keyof T; label?: string; required?: boolean };
+) & { name: Leaves<T> | Paths<T>; label?: string; required?: boolean };
```

Comment:
this needs more explanation
- what is Leaves and Paths, why do we need both?
- and: we really, really need a demo using this
    - there are already a few nested objects for product type
    - please do yourself and everyone else a favor and develop using demo

Reply:
I faced the problem with the demo-code when I was moving my implementation. (in my specific project the commented-out version did work, as there was no recursion or block-structure)

I tried to support this:

```
Entity: {
    foo: string;
    address: {
        city: string;
        zip: string;
    };
    block: {
        .... some properties
    }
}

const formConfig: {
    fields: [
        { name: "foo", ...},
        { name: "address.zip", ...}
        { name: "block", ... }
    ]
}
```

Leaves only supports "foo", "address.zip", "block.some_property"
Paths only supports "foo", "address", "block"

so I combined those two. I'll add an example for name: "packageDimensions.height" as this is the only case missing.

Would you say this is something that requires a comment next to Leaves | Paths ?

Reply:
why do you need Paths? `packageDimensions` is not a possible value, only `packageDimensions.height` is

Reply:
because of properties like [image-block](https://github.com/vivid-planet/comet/pull/1599/files#diff-4c7f0d1ce5a7a2c93622d806c1f65eaa598b3ac73a45f0035f982f2943c01b5dR19) and probably special form-fields supporting other object-structures. (but there may be other solutions like defining multiple properties, e.g. Pick/ShowCoordinates-Field need to set `{type: "coordinates", longitude: "place.lng", latitude: "place.lat"}` instead of `{type: "coordinates", name: "place"}`

Reply:
I also thought of make this configurable with separate generic parameter FormConfig<T, ObjectFields>

`const type MyConfig = FormConfig<Product, "image" | "place">;`

but I don't know how to implement this kind of typing-behaviour and another reason: if, as the developer, I'm required to define the special fields, it shouldn't be to much of a problem to just don't use them in the first place.

this is probably helpful if my entity is changed afterwards. but to support this we might need some way to detect, at least for now, any block-type.